### PR TITLE
feat(chain): Arweave integration

### PR DESCRIPTION
**Pull request by clearloop**
## Changes

- add arweave types

## Tests

- https://github.com/chainSafe/subgraph-arweave-example

### CODE REVIEW

**Review by maoueh — CHANGES_REQUESTED** [review]
Pretty good, main stuff that is missing is the [TypeId](https://github.com/graphprotocol/graph-ts/blob/main/global/global.ts#L15) changes.

Each type defined in here need to be listed in the `TypeId` and [id_of_type](https://github.com/graphprotocol/graph-ts/blob/main/global/global.ts#L159) needs to be updated.

The value of the different IDs must be in a specific range, we decided that each chain would have a range of the ids allocated to it, for example Arweave could have the range 2000 - 3000. See https://github.com/graphprotocol/graph-node/pull/3446.

**Comment by evaporei** [comment]
@clearloop you can apply the `TypeId` changes that you've made in `graph-node` so we can approve/merge it 🙂

**Review by evaporei — APPROVED** [review]

**Comment by clearloop** [comment]
> @clearloop sorry for taking a bit to review this
> 
> but I think the CI is failing because you should copy the new files in the `test.js` script: https://github.com/graphprotocol/graph-ts/blob/main/test/test.js#L6
> 
> I you can try running `npm test` locally please 🙂

```
$ yarn run v1.22.17
warning package.json: License should be a valid SPDX license expression
$ node test/test.js
Compiling test file test/bigInt.ts to WASM...
Compiling test file test/bytes.ts to WASM...
Compiling test file test/entity.ts to WASM...
Running "test/bigInt.ts" tests...
Running "testBigInt"...
Running "test/bytes.ts" tests...
Running "testBytesWithByteArray"...
Running "testBytesFromUTF8"...
Running "test/entity.ts" tests...
Running "testEntity"...
Done in 1.03s.
```

`npm test` works on my machine, but `npm build` doesn't

```
$ yarn build
warning package.json: License should be a valid SPDX license expression
$ asc --explicitStart --exportRuntime --runtime stub index.ts helper-functions.ts --lib graph-ts -b index.wasm
asc: unrecognized option '--explicitStart'
Advanced Strategic Command: a turn based strategy game 
usage: asc [ -xylcrwfqhv ] 
  [ -x ] [ --xresolution ]  (type=INTEGER, range=800..., default=1024)
         Set horizontal resolution to <X>
  [ -y ] [ --yresolution ]  (type=INTEGER, range=600..., default=800)
         Set vertical resolution to <Y>
  [ -l ] [ --load ]  (type=STRING)
         Load a map, save game, or email game on startup
  [ -c ] [ --configfile ]  (type=STRING)
         Use given configuration file
  [ -r ] [ --verbose ]  (type=INTEGER, range=0...10, default=0)
         Set verbosity level to x (0..10)
  [ -w ] [ --window ]  (type=FLAG)
         Disable fullscreen mode
  [ -f ] [ --fullscreen ]  (type=FLAG)
         Enable fullscreen mode (overriding config file)
  [ -q ] [ --nosound ]  (type=FLAG)
         Disable sound
  [ -h ] [ --help ]  (type=FLAG)
         Display help information.
  [ -v ] [ --version ]  (type=FLAG)
         Output version.
Done in 0.04s.
```

File: chain/arweave.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import '../common/eager_offset'
+import { Bytes } from '..'
```
Comment (evaporei) [inline chain/arweave.ts:2]: You just have to change this import line to make the `yarn build` work.

```typescript
import { Bytes } from '../common/collections'
```

Also you should rebase with the current `main` branch 😉

**Review by evaporei — CHANGES_REQUESTED** [review]

**Pull request merged**